### PR TITLE
Makes `flags_inv` better at actually hiding things when used on non-suit clothing

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

`flags_inv` contains a series of flags which dictate which items can be seen / which items are rendered when it is worn. This lets you hide things that shouldn't be visible, stops weird bugs

Right now it really doesn't work to hot. Most of the flags aren't checked so a lot of silly things happen.

Here's an example, which was broken for 2 reasons

image

1. it was using the wrong var
2. nowhere checked that the shoes were hidden

So, I went through and made the `update_x` procs consistently check when the slot is actually hidden.

image

Much better?

~~It's actually still broken. It's hiding the arms for some reason. But that's for another PR?~~

## Why It's Good For The Game

Flags that exist should work as expected.

## Changelog

:cl: Melbert
fix: Wedding dresses no longer render their shoes over the dress
fix: Things which have inv flags set should more consistently actually hide the things they mean to hide
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_update_icons.dm [hunk 1]
```diff
@@ -84,16 +84,14 @@ There are several things that need to be remembered:
 		var/obj/item/clothing/under/uniform = w_uniform
 		update_hud_uniform(uniform)
 
-		if(wear_suit && (wear_suit.flags_inv & HIDEJUMPSUIT))
+		if(check_obscured_slots(TRUE) & ITEM_SLOT_ICLOTHING)
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/human_update_icons.dm:87]: named param for true

File: code/modules/mob/living/carbon/human/human_update_icons.dm [hunk 2]
```diff
@@ -185,19 +182,18 @@ There are several things that need to be remembered:
 		add_overlay(bloody_overlay)
 	//Bloody hands end
 
-	var/mutable_appearance/gloves_overlay
 	if(gloves)
 		var/obj/item/worn_item = gloves
 		update_hud_gloves(worn_item)
-		var/icon_file
 
+		if(check_obscured_slots(TRUE) & ITEM_SLOT_GLOVES)
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/human_update_icons.dm:189]: same

File: code/modules/mob/living/carbon/human/human_update_icons.dm [hunk 3]
```diff
@@ -343,13 +334,12 @@ There are several things that need to be remembered:
 
 	if(s_store)
 		var/obj/item/worn_item = s_store
-		var/mutable_appearance/s_store_overlay
 		update_hud_s_store(worn_item)
 
-		s_store_overlay = worn_item.build_worn_icon(default_layer = SUIT_STORE_LAYER, default_icon_file = 'icons/mob/clothing/belt_mirror.dmi')
-
-		if(!s_store_overlay)
+		if(check_obscured_slots(TRUE) & ITEM_SLOT_SUITSTORE)
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/human_update_icons.dm:339]: same and everywhere else

**Review by Mothblocks — APPROVED** [review]

File: code/modules/mob/living/carbon/human/human_update_icons.dm [hunk 4]
```diff
@@ -479,18 +462,15 @@ There are several things that need to be remembered:
...
 
-			if(!(icon_exists(icon_file, RESOLVE_ICON_STATE(worn_item))))
-				icon_file = 'icons/mob/clothing/mask.dmi'
+		if(check_obscured_slots(transparent_protection = TRUE) & ITEM_SLOT_MASK)
+			return
 
-			mask_overlay = wear_mask.build_worn_icon(default_layer = FACEMASK_LAYER, default_icon_file = icon_file)
+		var/icon_file = 'icons/mob/clothing/mask.dmi'
+		if(!(icon_exists(icon_file, RESOLVE_ICON_STATE(worn_item))))
+			icon_file = 'icons/mob/clothing/mask.dmi'
```
Comment (Kylerace) [inline code/modules/mob/living/carbon/human/human_update_icons.dm:471]: this doesnt change anything

**Review by Kylerace — CHANGES_REQUESTED** [review]

**Pull request merged**